cl/sentinel/handlers/heartbeats.go

```diff
@@ -24,30 +24,50 @@ import (
 // Since packets are just structs, they can be resent with no issue
 
 func (c *ConsensusHandlers) pingHandler(s network.Stream) error {
+	peerId := s.Conn().RemotePeer().String()
+	if err := c.checkRateLimit(peerId, "ping", defaultRateLimits.pingLimit); err != nil {
+		return err
+	}
 	return ssz_snappy.EncodeAndWrite(s, &cltypes.Ping{
 		Id: c.metadata.SeqNumber,
 	}, SuccessfulResponsePrefix)
 }
 
 func (c *ConsensusHandlers) goodbyeHandler(s network.Stream) error {
+	peerId := s.Conn().RemotePeer().String()
+	if err := c.checkRateLimit(peerId, "goodbye", defaultRateLimits.goodbyeLimit); err != nil {
+		return err
+	}
 	return ssz_snappy.EncodeAndWrite(s, &cltypes.Ping{
 		Id: 1,
 	}, SuccessfulResponsePrefix)
 }
 
 func (c *ConsensusHandlers) metadataV1Handler(s network.Stream) error {
+	peerId := s.Conn().RemotePeer().String()
+	if err := c.checkRateLimit(peerId, "metadataV1", defaultRateLimits.metadataV1Limit); err != nil {
+		return err
+	}
 	return ssz_snappy.EncodeAndWrite(s, &cltypes.Metadata{
 		SeqNumber: c.metadata.SeqNumber,
 		Attnets:   c.metadata.Attnets,
 	}, SuccessfulResponsePrefix)
 }
 
 func (c *ConsensusHandlers) metadataV2Handler(s network.Stream) error {
+	peerId := s.Conn().RemotePeer().String()
+	if err := c.checkRateLimit(peerId, "metadataV2", defaultRateLimits.metadataV2Limit); err != nil {
+		return err
+	}
 	return ssz_snappy.EncodeAndWrite(s, c.metadata, SuccessfulResponsePrefix)
 }
 
 // TODO: Actually respond with proper status
 func (c *ConsensusHandlers) statusHandler(s network.Stream) error {
+	peerId := s.Conn().RemotePeer().String()
+	if err := c.checkRateLimit(peerId, "status", defaultRateLimits.statusLimit); err != nil {
```

Comment:
if the rate limit check fails send write to the stream []byte{0x2} and close it, this is per spec the status code for rate limiting